src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -2907,6 +2907,17 @@ private static CultureInfo GetCultureFromFormatProvider(IFormatProvider formatPr
             TypeConverter integerConverter = LanguagePrimitives.GetIntegerSystemConverter(resultType);
             try
             {
+                try
+                {
+                    return Convert.ChangeType(
+                        Parser.ScanNumber(strToConvert, resultType, shouldTryCoercion: false),
+                        resultType,
+                        System.Globalization.CultureInfo.InvariantCulture.NumberFormat);
+                }
+                catch (ParseException)
```

Comment:
Documentation says Convert.ChangeType() can throw InvalidCastException, FormatException, etc.  To prevent more breakage than necessary, it seems like we should catch a generic Exception exception here and let the original code path run.

Reply:
This is intentional (I should probably comment it).

The original code path has an outer `catch (Exception e)` block below, where the general exceptions from ChangeType() are handled. This isn't the _clearest_ code form, I'll readily admit, but it's the simplest way to handle both errors in the right way, I think.

If there's a parse exception, the method goes on to call the general `integerConverter.ConvertFrom()` method as it was before, but if there's a _general_ exception either there or during the `ChangeType()` an entirely different code path is taken... assuming I am understanding my try/catch logic there correctly. 😄

Reply:
What I mean is that the code now first tries the new `TryScanNumber()` method before reverting back to the old code path (`integerconverter` or `ChangeType` directly on `strToConvert`).  It seems possible that `TryScanNumber()` could throw an exception when the old code path(s) might have succeeded, causing a regression.  To be safe it seems to me it is best to catch generic `Exception` instead of `ParseException`  in `TryScanNumber()` to ensure fallback code paths always get a chance at the conversion.

Reply:
Oh, I see what you mean, I think. So the only case I can see that happening is if `ScanNumber()` succeeds in getting a value, but the subsequent `Convert.ChangeType()` fails for some reason... _and_ the subsequent code paths wouldn't have necessarily failed anyway on that error.

I would generally expect that if the `ChangeType()` fails, so too would `integerConverter`'s method anyway, but I suppose there are potential edge cases where this wouldn't go so well... better safe than sorry, and at least in theory it shouldn't have to do the double-check too often, so it should be OK.

Will change, thank you!